dxp/resource_serving.md

```diff
@@ -0,0 +1,161 @@
+# Resource Serving in DXP
+
+This document explains how different resources (`.js`, `.css`, etc.) are handled by DXP to be served to the browser.
+
+## Common Serving Architecture
+
+When it comes to serving anything from DXP there are two important files from which to start the investigation. Both are inside [portal-web/docroot/WEB-INF](https://github.com/liferay/liferay-portal/tree/master/portal-web/docroot/WEB-INF):
+
+1. [web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/web.xml): this is the standard `web.xml` file of DXP webapp. It defines the portlets and filters that should be loaded by the application server (by default Tomcat).
+2. [liferay-web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/liferay-web.xml): this is a clone of `web.xml` that is loaded by DXP once `com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter` is loaded.
+
+The default DXP's startup sequence goes like this:
+
+1. `Tomcat` is started.
```

Comment:
Can you instead trace this from the Big Bang onwards, so get the full story? 😂

Reply:
:thinking: It says `DXP startup sequence`. I'm not quite sure that can be classified as the startup itself, but more as a prerequisite needed for the startup to take place, being the startup a necessary cause of the Big Bang (probably).

However, we should keep an eye on this, in case something changes in the future, or new knowledge appears about the subject matter.